pkg/controller/jobs/rayjob/rayjob_controller.go

```diff
@@ -163,7 +163,8 @@ func (j *RayJob) Finished() (metav1.Condition, bool) {
 		Reason:  string(j.Status.JobStatus),
 		Message: j.Status.Message,
 	}
-	return condition, rayjobapi.IsJobTerminal(j.Status.JobStatus)
+
+	return condition, j.Status.JobStatus == rayjobapi.JobStatusFailed || j.Status.JobStatus == rayjobapi.JobStatusSucceeded
```

Comment:
I'm not sure if this is a bug or the specification on the kuberay side.
My concern is this behavior might be changed according to the kuberay versions.

Are there any documentation or API comments?

Reply:
it sounds like it should be fixed on the rayjob side.

Reply:
In kuberay, when stopping a job, it will check whether the job is *stopped*, *succeeded* or *failed*, if no, kuberay will stop the job, so `IsJobTerminal` includes *stop* status. But here, we should only consider the terminated status. The name is puzzling anyway.

Reply:
Ah, I see. The `stopped` means `suspended`.

Reply:
It still sounds like the name or the logic could be fixed in kuberay.